### PR TITLE
Adding test workflows, basic test setup with vitest, update README

### LINKED ISSUE #4: Add tests and more requirements for PR's
- Add basic unit tests
- Make sure PR's successfully run format/lint/test commands before merging

### PR DESCRIPTION
**Pull request by calebjacob**
Closes: https://github.com/near/pagoda-ui/issues/4

- Implement GitHub workflows for formatting, linting, checking types, running tests,
- Pull in Vitest and React Testing to set up structure for unit testing
- Added a single test suite for our debounce hook just to get us started - we'll continue to add more tests as we have time

### CODE REVIEW

File: src/hooks/debounce.test.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  afterEach(() => {
+    vi.clearAllTimers();
+  });
+
+  it('should debounce function', () => {
+    const myMethod = vi.fn(() => 0);
+    const { result } = renderHook(() => useDebouncedFunction(myMethod, 300));
+    result.current();
+    result.current();
```
Comment (shelegdmitriy) [inline src/hooks/debounce.test.ts:19]: Why is this called twice?

**Review by shelegdmitriy — APPROVED** [review]
Great job!

File: src/hooks/debounce.test.ts (hunk 1 above)
Comment (calebjacob) [reply src/hooks/debounce.test.ts:19]: Good question! It does look funny, but this is actually to make sure debounce is doing it's job (calling it twice, but making sure it's executed only once).

Comment (shelegdmitriy) [reply src/hooks/debounce.test.ts:19]: Ok, makes sense

**Pull request merged**